google/cloud/internal/grpc_opentelemetry.cc

```diff
@@ -30,6 +32,41 @@ namespace internal {
 
 #ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
 
+namespace {
+
+/**
+ * A [carrier] for gRPC.
+ *
+ * This code originates from the following example:
+ *
+ * https://github.com/open-telemetry/opentelemetry-cpp/blob/a343da043e1351c3cc3003bfce94724345ee22f1/examples/grpc/tracer_common.h#L26-L45
+ *
+ * [carrier]:
+ * https://opentelemetry.io/docs/reference/specification/context/api-propagators/#carrier
+ */
+class GrpcClientCarrier
+    : public opentelemetry::context::propagation::TextMapCarrier {
+ public:
+  explicit GrpcClientCarrier(grpc::ClientContext& context)
+      : context_(context) {}
+
+  // Unneeded by clients.
+  opentelemetry::nostd::string_view Get(
+      opentelemetry::nostd::string_view) const noexcept override {
+    return "";
```

Comment:
Even though our code is client-only, somebody may use it in a server.  Does that matter?  What if they are using the functions framework and get a HTTP request with all the right tracing headers?

Reply:
> Does that matter?

I don't think so. This `Get()` would be used to continue some span started in a Google server, that was passed through the client context. That could be a thing, but I don't think it is at the moment.

> What if they are using the functions framework and get a HTTP request with all the right tracing headers?

Customers would:
1. extract the span context from the incoming HTTP request
2. start a span using that context (i.e. the new span would be a child of the remote span)
3. set that span as active (so the `google-cloud-cpp` spans will be a child of that span)

(It is the "Extract headers to context" part of this: https://opentelemetry.io/docs/instrumentation/cpp/manual/#context-propagation)

I am sure an example of how to do this would be helpful to the customers. But that is several milestones down the line.

Reply:
SGTM